Bottom: 48c56127291d47740d32fad39d08aa9d0c6f7d7e
Top:    5d1fb77e4d7c1172e6e18fd89b62e4108958cb56
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-05-11 02:06:01 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 6cea843..e4612f4 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1160,7 +1160,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
